core/dbt/include/starter_project/dbt_project.yml

```diff
@@ -0,0 +1,38 @@
+
+# Name your project! Project names should contain only lowercase characters
+# and underscores. A good package name should reflect your organization's
+# name or the intended use of these models
+name: 'my_new_project'
+version: '1.0.0'
+config-version: 2
+
+# This setting configures which "profile" dbt uses for this project.
+profile: 'default'
```

Comment:
I would _love_ if we could figure out a way to automatically populate `name` and `profile` with the `project_name` argument supplied to `dbt init`. Right now, that `project_name` arg is just used to name the file directory, but not the actual project name. Pretty confusing! Also, we tend to discourage using a profile named `default`, and yet here we are...

I know all we're really doing here is `shutil.copytree`. Is there any sane way to try editing the files after copying?

For reference, I was just triaging a related issue, so I mentioned this over there as well: https://github.com/fishtown-analytics/dbt/issues/3462#issuecomment-865314756

Reply:
We have a lot of expertise with jinja templates :). Perhaps we could make a dbt_project jinja template and substitute the project name. Or just do a python substitution if it's simpler. I guess jinja would only make sense if there are other ways we could leverage it to customize the project file.

Reply:
Heh, you're not wrong. Back in the day, we did a lot of customization for client projects over here: https://github.com/fishtown-analytics/dbt-init

The purpose there was cross-applying some pretty opinionated configs for a given adapter. I think, for our purposes, we shouldn't go much further beyond `name` and `profile`.

Reply:
I'll take a crack at doing this in another PR to follow this one up. So we want `project_name` to be the `name` and `profile` field in the dbt_project.yml file (just to verify)?

Reply:
Yes, that's right!